db/migrate/20161109150714_add_templates_to_resolve_url.rb

```diff
@@ -0,0 +1,21 @@
+class AddTemplatesToResolveUrl < ActiveRecord::Migration[5.0]
+  def up
+    Agents::WebsiteAgent.find_each do |agent|
+      keys = agent.event_keys
+      next if keys.nil? || !keys.include?('url') || agent.options.key?('template')
+      agent.options['template'] = keys.each_with_object({}) { |key, template|
+        case key
+        when 'url'
+          template[key] = '{{ url | to_uri: _response_.url }}'
+        else
+          template[key] = "{{ #{key} }}"
```

Comment:
I changed the semantics of `template` before it was merged so that the result of `template` would become the final output instead of that of `extract` merged with it.  So, keys not in `template` will be dropped from created events.

This is because that would be what users might expect from the word `template`, and would allow users to extract as many parts as they want for building a new key without polluting the final payload with the intermediate keys.  On the other hand, I admit there are times when copying keys from `extract` to `template` makes you feel awkward and repeating yourself.  Should we improve this?